plugins/search/providers/elastic_search/scripts/aliases/ElasticIndexRotationWorker.php

```diff
@@ -189,6 +192,61 @@ protected function runRotate()
 		$aliasesToAdd[] = new ElasticIndexAlias($newIndex, $this->indexAlias);
 
 		$this->changeAliases($aliasesToAdd, $aliasesToRemove);
+
+		if($this->handleUnusedIndices)
+		{
+			$this->handleUnusedIndices();
+		}
+	}
+
+	protected function handleUnusedIndices()
+	{
+		$indicesToDelete = $this->getUnusedIndicesToHandle();
+		$this->deleteIndices($indicesToDelete);
+	}
+
+	protected function deleteIndices($indicesToDelete)
+	{
+		if(!$indicesToDelete)
+		{
+			KalturaLog::debug("No indices to delete");
+			return;
+		}
+
+		$indicesToDeleteString = implode(",", $indicesToDelete);
+		if($this->dryRun)
+		{
+			KalturaLog::debug("Dry run - deleting indices {$indicesToDeleteString}");
+		}
+		else
+		{
+			$response = $this->client->deleteIndex($indicesToDeleteString);
+			if (!$response)
+			{
+				die("Could not delete indices: {$indicesToDeleteString}\n");
+			}
+		}
+	}
+
+	protected function getUnusedIndicesToHandle()
+	{
+		list($currentIndexingIndices, $currentSearchingIndices) = $this->getCurrentStateMap();
+		$indicesToDelete = array();
+		$response = $this->client->getIndicesForIndexPattern($this->indexPattern . '*');
+		if (!$response)
+		{
+			die("Could not get indices info\n");
+		}
+
+		foreach ($response as $indexName => $arr)
+		{
+			if(!array_key_exists($indexName, $currentSearchingIndices))
```

Comment:
i think it would be better to also check that you don't have an alias set on the index